graphql_compiler/query_pagination/__init__.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from collections import namedtuple
+
+from graphql.language.printer import print_ast
+
+from graphql_compiler.cost_estimation.cardinality_estimator import estimate_number_of_pages
+from graphql_compiler.query_pagination.parameter_generator import (
+    hydrate_parameters_of_parameterized_query
+)
+from graphql_compiler.query_pagination.query_parameterizer import generate_parameterized_queries
+
+
+PaginationQueries = namedtuple(
+    'PaginationQueries',
+    (
+        'next_page_query',          # Document, AST for the query that along with
+                                    # next_page_parameters is expected to return the next page of
+                                    # page of results of the original query.
+        'next_page_parameters',     # dict, parameters with which to execute the next page query.
+        'continuation_query',       # Document or None, AST for the query that along with
+                                    # continuation_parameters returns the rest of the results of the
+                                    # original query, or None if the original query was not split
+                                    # into two queries.
+        'continuation_parameters',  # dict or None, parameters with which to execute the
+                                    # continuation query, or None if the continuation query was not
+                                    # generated.
+    )
+)
+
+
+def split_into_next_page_query_and_continuation_query(
+    schema_graph, statistics, query_ast, parameters, num_pages
+):
+    """Split a query into two equivalent queries, one of which will return roughly a page of data.
+
+    First, two parameterized queries are generated that contain filters usable for pagination i.e.
+    filters with which the number of results can be constrained. Parameters for these filters are
+    then generated such that one of the new queries will return roughly a page of results, while the
+    other query will generate the rest of the results. This ensures that the two new queries' result
+    data is equivalent to the original query's result data.
```

Comment:
This is too much internal methodology for a function in `__init__.py`. You can put it in a different module, and make a function in `__init__.py` that just directly calls it.